### PR TITLE
Separate internal and external TLS config

### PR DESCRIPTION
**Pull request by mhutter**
Part of #80

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+== Changes
+
+Configuration of TLS aspect has been overhauled:
+* Separation of internal (ingress to Keycloak) and external (client to ingress) TLS configuration
+
+
+
+== Step-by-step guide
+
+When upgrading the component, the following changes to the inventory are required:app-name:
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc:14]:
```suggestion
When upgrading the component, the following changes to the inventory are required:
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -94,7 +94,18 @@ The Keycloak pod replicas.
 Usualy two for the redundancy during the maintenance.
 
 
-== `tls.provider`
+== `tls`
+
+[horizontal]
+type:: dict
+default:: _see below_
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:101]: If this section is just there as an "introduction" to other `tls.*` parameters, I guess we can remove this?
```suggestion
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -227,7 +230,43 @@ The namespace where the ingress controller is running.
 This is only relevant when enabling the network policy with `helm_values.networkPolicy.enabled`.
 
 
-== `ingress.tls.secretName`
+== `ingress.tls`
+
+[horizontal]
+type:: dict
+default:: _see below_
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:237]: Same here
```suggestion
```

File: docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc (hunk 1 above)
Comment (mhutter) [reply docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc:14]: My editor trying to be a smart-ass, gets me every time 😂

**Review by ccremer — APPROVED** [review]

**Comment by ccremer** [comment]
The how-to page isn't linked in the navigation. Is this expected to come later in other PRs?

**Comment by mhutter** [comment]
I totally forgot it, but will add later once I'm done with my shenanigans

**Pull request merged**